core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,323 @@
+ * @param {string = null} [barValueColor] - The color the values are shown in (default = no values shown)
+ * @param {number = 4} [numberTicksY] - Number of Ticks on the y-axis (will deviate by 1 if the values wouldn't make a nice scale otherwise)
+ */
+export class HorizontalBarChart extends Chart {
+  constructor({
+    selector,
+    data,
+    barColor = "blue",
+    width = 400,
+    height = 400,
+    barValueColor,
+    //barWidth is actually bar height since it is a horizontal bar chart but for all intents and purposes it is the bar's width
+    barWidth,
+    numberTicksX,
+    groups,
+  }) {
+    super({ selector, data, width, height, margin });
+    this._groups = groups;
+    this._barColor = barColor || "blue";
+    this._barWidth = barWidth;
+    this._xScale = d3.scaleLinear().range([0, this.chartWidth]);
+    this._yScales = this._getYscales(groups);
+    this._barValueColor = barValueColor;
+    this._numberTicksX = numberTicksX || 4;
+  }
+
+  _getYscales(groups) {
+    if (groups) {
+      const yScales = [];
+      let headingSpace = groupHeadlineHeight + headlinePadding;
+      for (let group of groups) {
+        const relevantBars = this.data.filter(
+          (data) => data.group === group.id
+        ).length;
+        // totalSpan * barPadding = paddingSpan
+        const barsSpan = relevantBars * this._barWidth;
+        const paddingSpan = barsSpan / (1 / barPadding - 1);
+        const totalSpan = barsSpan + relevantBars * paddingSpan;
+        // const totalSpan = barsSpan;
+        const scale = d3
+          .scaleBand()
+          .range([headingSpace, totalSpan])
+          .padding(barPadding);
+        headingSpace = headingSpace + totalSpan;
+        yScales.push({ scale, id: group.id });
+      }
+      return yScales;
+    } else {
+      return [
+        {
+          scale: d3
+            .scaleBand()
+            .range([0, this.chartHeight])
+            .padding(barPadding),
+        },
+      ];
+    }
+  }
+
+  _adaptScalesToData() {
+    this._xScale.domain([0, d3.max(this.data, (d) => d.value)]);
+    for (let scale of this._yScales) {
+      const relevantData = scale.id
+        ? this.data
+            .filter((data) => data.group === scale.id)
+            .sort((a, b) => b.value - a.value)
+            .map((data) => data.title)
+        : this.data.sort((a, b) => b.value - a.value).map((data) => data.title);
+      scale.scale.domain(relevantData);
+    }
+  }
+
+  _updateExistingBars(bars) {
+    bars
+      .transition()
+      .duration(750)
+      .attr("x", 0)
+      .attr("width", initializingBarHeight)
+      .attr("y", (d) => {
+        for (let scale of this._yScales) {
+          if (d.group === scale.id)
+            return this._barWidth
+              ? scale.scale(d.title) +
+                  (scale.scale.bandwidth() - this._barWidth) / 2
+              : scale.scale(d.title) + scale.scale.bandwidth() / 4;
+        }
```

Comment:
This is a recurring Theme. Maybe a function that you pass a function and it wraps it with the for and if? Is fine without just an idea